modules/javafx.graphics/src/main/native-font/directwrite.cpp

```diff
@@ -845,6 +845,13 @@ jobject newD2D1_MATRIX_3X2_F(JNIEnv *env, D2D1_MATRIX_3X2_F *lpStruct)
 /*                                                                        */
 /**************************************************************************/
 
+static void WICImagingFactoryFlsCallback(void *data)
+{
+    IWICImagingFactory *iwicImagingFactory = (IWICImagingFactory *)data;
+    iwicImagingFactory->Release();
```

Comment:
I don't think this is correct. According to the documentation for `CoCreateInstance`, it already does the Release so it should not be done again. Can you comment on this?

Reply:
I guess you are referring to this paragraph in the documentation?

> The CoCreateInstance function provides a convenient shortcut by connecting to the class object associated with the specified CLSID, creating an uninitialized instance, and releasing the class object. As such, it encapsulates the following functionality:
> ```
> CoGetClassObject(rclsid, dwClsContext, NULL, IID_IClassFactory, &pCF); 
> hresult = pCF->CreateInstance(pUnkOuter, riid, ppvObj) 
> pCF->Release(); 
> ```

From my understanding, what it does is that it releases the *class factory* (`pCF`). The instance created with `pCF->CreateInstance` still has a non-zero reference count and so `Release()` still needs to be called on it.

The code examples in https://docs.microsoft.com/en-us/windows/desktop/learnwin32/managing-the-lifetime-of-an-object show `Release()` being called on objects created with `CoCreateInstance`, so it should be correct.

Reply:
Yes, I think you are correct.